static/src/javascripts/projects/commercial/modules/reload-page-on-consent-change.spec.ts

```diff
@@ -0,0 +1,66 @@
+import { onConsentChange } from '@guardian/consent-management-platform';
+import type {
+	Callback,
+	ConsentState,
+} from '@guardian/consent-management-platform/dist/types';
+import { _ } from './reload-page-on-consent-change';
+
+// we need to bypass the `once()` wrapper for testing
+const { _reloadPageOnConsentChange: reloadPageOnConsentChange } = _;
+
+jest.mock('@guardian/consent-management-platform', () => ({
+	onConsentChange: jest.fn(),
+}));
+
+const mockOnConsentChange = (consentState: ConsentState) =>
+	(onConsentChange as jest.Mock).mockImplementation((cb: Callback) =>
+		cb(consentState),
+	);
+
+const reload = jest.fn();
+
+jest.spyOn(window, 'location', 'get').mockImplementation(
+	() => ({ reload } as unknown as Location),
+);
+
+describe('reloadPageOnConsentChange()', () => {
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('reloads if the value of canTarget changes from true to false', () => {
+		const initialConsentState: ConsentState | undefined = {
+			canTarget: true,
+			framework: 'tcfv2',
+		};
+		const consentState: ConsentState = {
+			canTarget: false,
+			framework: 'tcfv2',
+		};
+
+		mockOnConsentChange(initialConsentState);
+		reloadPageOnConsentChange();
+		mockOnConsentChange(consentState);
+		reloadPageOnConsentChange();
```

Comment:
It would be great if we had a way to simulate consent state changing i.e. a function like `setConsentState(initialConsentState)` that would trigger the listeners registered with `onConsentChange`.
That would allow us to write a test that reflects the real-life scenario a bit more closely like
```
reloadPageOnConsentChange(); // only called once, on page load
setConsentState(initialConsentState); // initial 
setConsentState(consentState); // change
```